### PR TITLE
Define NEL in terms of Fetch

### LINKED ISSUE #54: Interaction with ServiceWorker
> Has any thought been given to if/when SWs interact with reports? - https://github.com/w3ctag/spec-reviews/issues/24#issuecomment-104466784

### PR DESCRIPTION
**Pull request by igrigorik**
Closes #54.

### CODE REVIEW

**Comment by toddreifsteck** [comment]
@annevk could you please take a look?

File: index.html [hunk 1]
```diff
@@ -363,13 +363,27 @@
...
+          <li>Let <dfn>req</dfn> be a new <a href="https://fetch.spec.whatwg.org/#concept-request">request</a>, initialized as follows:</li>
 
-          <li>If the fetch is successful (2xx HTTP response code), the user agent MUST abort the remaining steps.</li>
+          <ul>
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-url">url</a> to <a title="report-uri">report URL</a></li>
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-method">method</a> to <code>POST</code></li>
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-header-list">header list</a> <code>Content-Type</code> value to <code>application/nel-report</code></li>
```
Comment (annevk) [inline index.html]: You want to append to a header list. It doesn't have `Content-Type` by default.

File: index.html [hunk 2]
```diff
@@ -363,13 +363,27 @@
...
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-body">request body</a> to <a>report body</a></li>
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-referrer-policy">referrer policy</a> to <code>origin-only</code></li>
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-credentials-mode">credentials mode</a> to <code>same-origin</code></li>
+            <li>Set <a href="https://fetch.spec.whatwg.org/#concept-request-context">context</a> to <code>nelreport</code></li>
```
Comment (annevk) [inline index.html]: There's no context.

File: index.html [hunk 3]
```diff
@@ -363,13 +363,27 @@
...
 
-          <li>If the fetch failed with a <a href="https://tools.ietf.org/html/rfc7231#section-6.5.9">410 Gone</a> HTTP response code ([[RFC7231]]), the user agent MUST update the <a>NEL policy</a> by removing the current <a title="report-uri">report URL</a> from the <a>set of report URLs</a>.</li>
+          <li><a href="http://www.w3.org/html/wg/drafts/html/CR/webappapis.html#queue-a-task">Queue a task</a> to <a href="https://fetch.spec.whatwg.org/#concept-fetch">Fetch</a> <a>req</a>. Let the <a href="https://www.w3.org/TR/html5/webappapis.html#task-source">task source</a> for these <a href="https://www.w3.org/TR/html5/webappapis.html#concept-task">tasks</a> to be the <dfn data-dfn-type="dfn" id="dfn-network-error-logging-task-source">Network Error Logging task source.</dfn></li>
 
-          <li>If the fetch failed after multiple delivery attempts the user agent MAY update the <a>NEL policy</a> by removing the current <a title="report-uri">report URL</a> from the <a>set of report URLs</a>.
+          <li>If the <a href="https://fetch.spec.whatwg.org/#concept-response-status">response status</a> is a 2xx response code, the user agent MUST abort the remaining steps.</li>
```
Comment (annevk) [inline index.html]: You can use "ok status" here.

**Comment by igrigorik** [comment]
Closing, as this is no longer relevant. The intent is to have NEL be defined on top of Reporting API: https://github.com/WICG/network-error-logging/issues/67

**Pull request closed without merging**